.pre-commit-config.yaml

```diff
@@ -15,49 +17,20 @@ repos:
       # - id: end-of-file-fixer
       # - id: trailing-whitespace
 
-  - repo: https://github.com/pre-commit/pygrep-hooks
-    rev: v1.10.0
-    hooks:
-      - id: rst-directive-colons
-      - id: rst-inline-touching-normal
-      - id: text-unicode-replacement-char
-
-  # - repo: https://github.com/codespell-project/codespell
-  #   rev: v2.2.5
-  #   hooks:
-  #     - id: codespell
-  #       args: ["--write-changes"]
-  #       additional_dependencies:
-  #         - tomli
-
   - repo: https://github.com/astral-sh/ruff-pre-commit
-    rev: "v0.1.5"
+    rev: 'v0.6.8'
     hooks:
       - id: ruff
-        args: ["--fix", "--show-fixes"]
+        args: ["--fix"]
       # - id: ruff-format
 
-  - repo: https://github.com/MarcoGorelli/cython-lint
-    rev: v0.15.0
+  - repo: https://github.com/PyCQA/bandit
+    rev: 1.7.10
     hooks:
-      - id: cython-lint
-      # - id: double-quote-cython-strings
+      - id: bandit
+        args: ["-r", "-ll", "src", "tests"]
```

Comment:
Why is this being added? Ruff can do this check, https://docs.astral.sh/ruff/rules/#flake8-bandit-s.

Reply:
This was added to align the checks with jwst which uses bandit:
https://github.com/spacetelescope/jwst/blob/ffa02661593f4c53d66ab17fa837aa72a358def6/.pre-commit-config.yaml#L28

Reply:
No objection from me but that's beyond the scope of this PR.